### PR TITLE
chore(core): integrate monsoon-openstack-auth gem as elektra plugin

### PR DESCRIPTION
**Pull request by andypf**
This PR integrates the `monsoon-openstack-auth` gem directly into the elektra application as a plugin to streamline maintenance.

## Background

Previously, the authentication functionality was maintained as a separate gem, which created several challenges:
- Dependency version conflicts during bundle updates  
- Duplicate maintenance effort across repositories

## Solution

Instead of decomposing the gem into individual components, we integrated the entire gem as an elektra plugin for simpler maintenance and better integration.

### 1. Plugin Integration
- **Moved gem to `plugins/monsoon-openstack-auth/`** 
- Cloned complete gem structure maintaining all functionality
- Preserved existing module structure and namespacing

### 2. Test Suite Modernization  
- **Removed FactoryGirl dependency** - replaced with native RSpec doubles and stubs
- **Fixed deprecated test syntax** - updated to RSpec 3.13 standards
- **Resolved broken authentication stubs** - ensured consistent test state
- **Added comprehensive test coverage** for edge cases and error conditions

### 3. Test Execution Order Fix
- **Implemented custom RSpec ordering** to run authentication tests first
- **Prevents authentication stub pollution** between plugin tests
- **Resolves flaky test issues** caused by other plugins (lbaas2, email) modifying authentication state

### CODE REVIEW

**Comment by andypf** [comment]
self review done!

**Review by ArtieReus — CHANGES_REQUESTED** [review]
I think this PR still need a bit cleanup, but it would be fine for me to do it in a second interation. The only change request is the comment with the eval...

File: plugins/monsoon-openstack-auth/app/assets/images/monsoon_openstack_auth/.keep
[the remark this answers is not in the document]
Comment (andypf) [reply plugins/monsoon-openstack-auth/app/assets/images/monsoon_openstack_auth/.keep:1]: good point, I will remove it

File: plugins/monsoon-openstack-auth/lib/monsoon_openstack_auth/connection_driver/default.rb [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      end
+
+      protected
+
+        def cache
+          impl = MonsoonOpenstackAuth.configuration.token_cache
+          impl.new
+        end
+
+        def filter_params(params,filters=[:password])
```
[the remark this answers is not in the document]
Comment (andypf) [reply plugins/monsoon-openstack-auth/lib/monsoon_openstack_auth/connection_driver/default.rb:182]: What do you mean? Application credentials?

File: plugins/monsoon-openstack-auth/monsoon-openstack-auth.gemspec [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  s.summary     = "Authenticate against Openstack Keystone."
+  s.description = "This gem enables authentication for Ruby on Rails applications against Openstack Keystone Service using the Identity API v3."
+
+  s.files = Dir["{app,config,lib}/**/*", "MIT-LICENSE", "Rakefile", "README.rdoc"]
+  s.test_files = Dir["spec/**/*"]
+
+  s.add_dependency 'hashie'
+  s.add_dependency 'uglifier', '>= 1.3.0'
+  s.add_dependency 'rails'
+  s.add_dependency 'excon'
```
[the remark this answers is not in the document]
Comment (andypf) [reply plugins/monsoon-openstack-auth/monsoon-openstack-auth.gemspec:19]: We do. Phase 2!

**Comment by andypf** [comment]
> I think this PR still need a bit cleanup, but it would be fine for me to do it in a second interation. The only change request is the comment with the eval...

Many thanks. You are right it needs a clean up. Since it is a core functionality my plan was just move it to elektra and update dependencies in the first iteration.

**Review by hgw77 — APPROVED** [review]

**Review by edda — APPROVED** [review]
